scenario/scenario.go

```diff
@@ -11,465 +14,352 @@ import (
 	"github.com/0xProject/0x-mesh/ethereum"
 	"github.com/0xProject/0x-mesh/ethereum/signer"
 	"github.com/0xProject/0x-mesh/ethereum/wrappers"
+	"github.com/0xProject/0x-mesh/scenario/orderopts"
 	"github.com/0xProject/0x-mesh/zeroex"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/stretchr/testify/require"
 )
 
-var ganacheAddresses = ethereum.GanacheAddresses
+var (
+	ethClient        *ethclient.Client
+	ganacheAddresses = ethereum.GanacheAddresses
+	ZRXAssetData     = common.Hex2Bytes("f47261b0000000000000000000000000871dd7c2b4b25e1aa18728e9d5f2af4c4e431f5c")
+	WETHAssetData    = common.Hex2Bytes("f47261b00000000000000000000000000b1ba0af832d7c05fd64161e0db78e85978e8082")
+)
 
-// CreateZRXForWETHSignedTestOrder creates a valid 0x orders where the maker wishes to trade ZRX for WETH
-func CreateZRXForWETHSignedTestOrder(t *testing.T, ethClient *ethclient.Client, makerAddress, takerAddress common.Address, wethAmount *big.Int, zrxAmount *big.Int) *zeroex.SignedOrder {
-	expirationTime := time.Now().Add(24 * time.Hour)
-	return createZRXForWETHSignedTestOrder(t, ethClient, makerAddress, takerAddress, wethAmount, zrxAmount, expirationTime)
+func init() {
```

Comment:
This was an intentional design decision. Setting up the maker state by default is unsafe because it can lead to lingering state between tests if it was not intended. I.e. if we have a test which does not use `BlockchainLifecycle` to reset the state, the state will linger and potentially interfere with other tests.

In addition, there is a performance impact to setting up the maker state when it is not necessary. There are many places in the existing tests where we don't need to set up the maker state (e.g. in __order_validator_test.go__ we have some tests for offchain validation). We just need a valid signed order. One of the goals of this PR was to serve those use cases more efficiently.

Reply:
That is a fair point. My point is that this way it is likely that we redundantly set up state when we didn't need to because the onus is now on the test writer to setup state, and it's likely that they do it for every default order they create (which is not always necessary). 

Having said this, I'm fine with dropping this suggestion. The argument that setting up this state is a leaky abstraction invalidates how nice I thought this would be.